azurerm/internal/services/storage/resource_arm_hpc_cache_nfs_target.go

```diff
@@ -0,0 +1,288 @@
+	defer cancel()
+
+	log.Printf("[INFO] preparing arguments for Azure HPC Cache NFS Target creation.")
+	name := d.Get("name").(string)
+	resourceGroup := d.Get("resource_group_name").(string)
+	cache := d.Get("cache_name").(string)
+
+	if d.IsNewResource() {
+		resp, err := client.Get(ctx, resourceGroup, cache, name)
+		if err != nil {
+			if !utils.ResponseWasNotFound(resp.Response) {
+				return fmt.Errorf("Error checking for existing HPC Cache NFS Target %q (Resource Group %q): %+v", name, resourceGroup, err)
+			}
+		}
+
+		if !utils.ResponseWasNotFound(resp.Response) {
+			return tf.ImportAsExistsError("azurerm_hpc_cache_nfs_target", *resp.ID)
+		}
+	}
+
+	hostName := d.Get("host_name").(string)
+	usageModel := d.Get("usage_model").(string)
+	namespaceJunctions := expandNamespaceJunctions(d.Get("namespace_junction").(*schema.Set).List())
+
+	// Construct parameters
+	param := &storagecache.StorageTarget{
+		StorageTargetProperties: &storagecache.StorageTargetProperties{
+			Junctions:  namespaceJunctions,
+			TargetType: storagecache.StorageTargetTypeNfs3,
+			Nfs3: &storagecache.Nfs3Target{
+				Target:     &hostName,
+				UsageModel: &usageModel,
+			},
+		},
+	}
+
+	future, err := client.CreateOrUpdate(ctx, resourceGroup, cache, name, param)
+	if err != nil {
+		return fmt.Errorf("Error creating HPC Cache NFS Target %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	if err := future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		return fmt.Errorf("Error waiting for creation of HPC Cache NFS Target %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	read, err := client.Get(ctx, resourceGroup, cache, name)
+	if err != nil {
+		return fmt.Errorf("Error retrieving HPC Cache NFS Target %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	if read.ID == nil {
+		return fmt.Errorf("Error retrieving HPC Cache NFS Target %q (Resource Group %q): `id` was nil", name, resourceGroup)
+	}
+
+	d.SetId(*read.ID)
+
+	return resourceArmHPCCacheNFSTargetRead(d, meta)
+}
+
+func resourceArmHPCCacheNFSTargetRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Storage.StorageTargetsClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parsers.HPCCacheTargetID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resp, err := client.Get(ctx, id.ResourceGroup, id.Cache, id.Name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[DEBUG] HPC Cache NFS Target %q was not found (Resource Group %q, Cahe %q) - removing from state!", id.Name, id.ResourceGroup, id.Cache)
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error retrieving HPC Cache NFS Target %q (Resource Group %q, Cahe %q): %+v", id.Name, id.ResourceGroup, id.Cache, err)
+	}
+
+	d.Set("name", id.Name)
+	d.Set("resource_group_name", id.ResourceGroup)
+	d.Set("cache_name", id.Cache)
+
+	if resp.StorageTargetProperties == nil {
```

Comment:
could we invert this for consistency with existing resource code?
```suggestion
	if props = resp.StorageTargetProperties; props != nil {
	...
```

Reply:
Sorry being out of format here.